lib/core/monitor_impl.toit

```diff
@@ -88,6 +88,7 @@ class __Monitor__:
           if deadline and Time.monotonic_us >= deadline: return false
         // Wait until notified. When we get back the monitor might be owned by
         // someone else.
+        if timer: timer.arm deadline
         await_ self
     finally:
       if timer: self.release_timer_ timer
```

Comment:
Perhaps move the null check inside release_timer_ ?

Reply:
Keeping as is for consistency with the other 'if timer' checks.